python_wikibase/data_types/time.py

```diff
@@ -0,0 +1,76 @@
+import collections
+import datetime
+
+from python_wikibase.data_types.data_type import DataType
+
+
+class DataTypeException(BaseException):
+    pass
+
+# Largely inspired from  https://github.com/dahlia/wikidata/blob/master/wikidata/datavalue.py
+
+
+class Time(DataType):
+    def __init__(self, py_wb, api, language):
+        super().__init__(py_wb, api, language)
+        self.value = None
+
+    def __str__(self):
+        return self.value.isoformat()
+
+    def unmarshal(self, data_value):
+        value = data_value["value"]
+
+        # See https://www.mediawiki.org/wiki/Wikibase/DataModel/JSON#time
+        # for property to extract from data_value
+
+        time = value['time']
+        # Strip '+' at beginning of string and 'Z' at the end
+        time = time[1:-1]
+
+        tz = value['timezone']
+        # Don't take in account 'before' and 'after' property
+        precision = value['precision']
+
+        if precision == 11:
+            self.value = datetime.date.fromisoformat(time[:-9])
+        elif precision == 14:
+            val = datetime.datetime.fromisoformat(time)
+            val.replace(tzinfo=datetime.timezone(offset=datetime.timedelta(minutes=tz)))
```

Comment:
I think that the `replace` method does not modifies the date, so you should do `val = val.replace(...)`

Reply:
You're right, it's now fixed.